test/bin/scenario.sh

```diff
@@ -194,6 +194,19 @@ launch_vm() {
 
     echo "Creating ${full_vmname}"
     local -r vm_wait_timeout=$(( VM_BOOT_TIMEOUT / 60 ))
+    local -r vm_pool_name="${VM_POOL_BASENAME}-${SCENARIO}"
+    local -r vm_pool_dir="${VM_DISK_BASEDIR}/${vm_pool_name}"
+
+    # Create the pool if it does not exist
+    if [ ! -d "${vm_pool_dir}" ] ; then
+        mkdir -p "${vm_pool_dir}"
+    fi
+    if ! sudo virsh pool-info "${vm_pool_name}" &>/dev/null; then
```

Comment:
Nit  - Should the `action_create` function be symmetric with `action_cleanup` in pool management? I see test/bin/manage_hypervisor_config.sh deletes all pools itself which makes sense for CI, but less so in a dev env where I would be executing `scenario.sh {create|cleanup} ...`.

Reply:
Yes, this makes a lot of sense. I will unhold this PR to unblock other merges and implement this suggestion separately in no time. 
/unhold